prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -151,7 +152,7 @@ def _build_llm(self) -> BaseChatModel:
         #  https://github.com/gnosis/prediction-market-agent/issues/107)
         llm = ChatOpenAI(
             model=self.model,
-            api_key=keys.openai_api_key.get_secret_value(),
+            api_key=SecretStrV1(keys.openai_api_key.get_secret_value()),
```

Comment:
Is this used like this in all instances of `APIKeys().openai_api_key`?

If so, is `pydantic.types.SecretStr` not a valid type?
- If it is, you could just do `api_key=keys.openai_api_key`.
- If not, you could change the return type of `APIKeys.openai_api_key` to `pydantic.v1.types.SecretStr`

Reply:
implemented 2nd approach